app/synchronizationmanager.cpp

```diff
@@ -8,21 +8,36 @@
  ***************************************************************************/
 
 #include "synchronizationmanager.h"
+#include "activeprojectmanager.h"
 
-SynchronizationManager::SynchronizationManager( MerginApi *backend, QObject *parent )
+SynchronizationManager::SynchronizationManager(
+  MerginApi *backend,
+  QObject *parent
+)
   : QObject( parent )
-  , mAutosyncController( nullptr )
   , mBackend( backend )
 {
-  QObject::connect( mBackend, &MerginApi::syncProjectFinished, this, &SynchronizationManager::syncProjectFinished );
-  QObject::connect( mBackend, &MerginApi::syncProjectStatusChanged, this, &SynchronizationManager::syncProjectStatusChanged );
-}
-
-SynchronizationManager::~SynchronizationManager()
-{
+  if ( mBackend )
+  {
+    QObject::connect( mBackend, &MerginApi::syncProjectFinished, this, &SynchronizationManager::syncProjectFinished );
+    QObject::connect( mBackend, &MerginApi::syncProjectStatusChanged, this, &SynchronizationManager::syncProjectStatusChanged );
+  }
 
+  QObject::connect( this, &SynchronizationManager::autosyncAllowedChanged, this, [this]( bool allowed )
+  {
+    if ( allowed )
+    {
+      this->setupAutosync();
```

Comment:
whatabout create `this->enableAutosync(boolean)` function?

Reply:
Outdated now due to refactoring